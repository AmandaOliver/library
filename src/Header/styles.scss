.header {
    margin: 0 0 20px 0;
    animation: pop-in .8s ;
    background-color: #3fbfbb;
    width: 100%;
    display: inline-flex;
    flex-flow: wrap row;
    justify-content: center;
    box-shadow: 0 5px 5px #e4e4e4a8;

    > span {
        align-self: center;
        font-family: "Circular-Pro-Black";
        font-size: 50px;
        color: white;
    }
}

@keyframes pop-in {
    0% {
        width: 0;
        opacity: 0;
        transform: translateY(-4em) scale(.8)
    }
    100% {
        width:100%;
        opacity: 1;
        transform: none;
    }
}
