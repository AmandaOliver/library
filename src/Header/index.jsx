import React, { PureComponent } from 'react';
import Search from '../Search'
import { worker, isLibraryLoaded } from '../library'
import './styles.scss'

class Header extends PureComponent {
  constructor() {
    super()
    this.state = {
      loaded: false
    }
    worker.addEventListener('message', this.updateFlag.bind(this))
  }
  updateFlag() {
    if (isLibraryLoaded()) {
      this.setState({
        loaded: true
      })
    }

  }
  render() {
    return (
      <header className="header">
        <span className="header__title">THE LIBRARY</span>
        {!this.state.loaded ? 'loading...' : <Search />}
      </header>
    )
  }
}

export default Header;
