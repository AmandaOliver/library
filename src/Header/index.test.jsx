import React from 'react'
import renderer from 'react-test-renderer';
import Header from '.'

describe('Header', () => {
  it('renders', () => {
    const component = renderer.create(
      <Header/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
