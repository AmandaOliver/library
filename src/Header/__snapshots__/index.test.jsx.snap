// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<header
  className="header"
>
  <span
    className="header__title"
  >
    THE LIBRARY
  </span>
  <div
    className="header__loading"
  >
    Loading
  </div>
</header>
`;
